### PR TITLE
DOCS-3296 Adds the Synthetics CircleCI Orb README into Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds single sourcing logic for CircleCI Orb documentation. 

### Motivation

DOCS-3296

### Preview

https://docs-staging.datadoghq.com/alai97/single-sourcing-ci-orb-docs/synthetics/cicd_integrations/circleci_orb/

### Additional Notes

Remember to add single sourcing logic to the `pull_config.yaml` file before merging! 

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by peculiarism — APPROVED** [review]

File: content/en/synthetics/cicd_integrations/_index.md
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/_index.md:43]:
```suggestion
    {{< nextlink href="synthetics/cicd_integrations/circleci_orbs" >}}CircleCI Orb{{< /nextlink >}}
```

File: local/bin/py/build/configurations/pull_config_preview.yaml
Comment (peculiarism) [inline local/bin/py/build/configurations/pull_config_preview.yaml:302]:
```suggestion
          title: Synthetics CI CircleCI Orb
```

File: local/bin/py/build/configurations/pull_config_preview.yaml
Comment (peculiarism) [inline local/bin/py/build/configurations/pull_config_preview.yaml:299]: Can we change the md file to `orb`?

**Review by peculiarism — CHANGES_REQUESTED** [review]
Just a couple of changes but looks good to go!

**Review by apigirl — APPROVED** [review]
This LGTM pending Beths suggestions. Let me know on slack if you need another review/approval

File: config/_default/menus/menus.en.yaml
Comment (alai97) [inline config/_default/menus/menus.en.yaml:2315]:
```suggestion
  - name: CircleCI Orb
    url: /synthetics/cicd_integrations/circleci_orb
    parent: synthetics_cicd_integrations
    identifier: synthetics_cicd_integrations_circleci_orb
```

File: Makefile
Comment (alai97) [inline Makefile:119]:
```suggestion
	@if [ content/en/synthetics/cicd_integrations/circleci_orb.md ]; then \
	rm -f content/en/synthetics/cicd_integrations/circleci_orb.md ;fi
```

File: content/en/synthetics/cicd_integrations/_index.md
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/_index.md:43]: Do we need to change the href?

Comment (alai97) [reply content/en/synthetics/cicd_integrations/_index.md:43]:
```suggestion
    {{< nextlink href="synthetics/cicd_integrations/circleci_orb" >}}CircleCI Orb{{< /nextlink >}}
```

**Review by peculiarism — APPROVED** [review]

**Pull request merged**
